raw_section_snooper_vocab.py

```diff
@@ -0,0 +1,138 @@
+    Appends the extracted information to the global vocab_codes DataFrame.
+    """
+    section_elements = tree.findall(expr, ns)
+
+    # Use the global vocab_codes DataFrame
+    global vocab_codes
+
+    # Loop through each element matching the expression
+    for section_element in section_elements:
+        
+        # Extract relevant attributes from the document
+        tag = re.sub(r'{.*}', '', section_element.tag)
+        code = section_element.get('code')
+        codeSystem = section_element.get('codeSystem')
+        codeSystemName = section_element.get('codeSystemName')
+        displayName = section_element.get('displayName')
+        templateId = section_element.get('templateId')
+
+        # Print extracted details for debugging or logging purposes
+        print((f"\nSECTION  {type(section_element)} "
+               f" tag:{tag} "
+               f" code: {code} "
+               f" codeSystem: {codeSystem} "
+               f" codeSystemName: {codeSystemName} "
+               f" displayName: {displayName} "
+               f" templateId: {templateId}"))
+
+        # Create a new DataFrame row
+        new_row = pd.DataFrame([{
+            'tag': tag,
+            'code': code,
+            'codeSystem': codeSystem,
+            'codeSystemName': codeSystemName,
+            'displayName': displayName,
+            'templateId': templateId
+        }])
+
+        # Append the new row to vocab_codes using pd.concat for better performance
+        vocab_codes = pd.concat([vocab_codes, new_row], ignore_index=True)
+
+def main():
+    """
+    Main function that parses arguments, processes the XML file, extracts code elements, and cleans up the DataFrame.
+    """
+    parser = argparse.ArgumentParser(
+        prog='CCDA - OMOP Code Snooper',
+        description="Finds all code elements and shows what concepts they represent",
+        epilog='epilog?')
+    parser.add_argument('-f', '--filename', required=True, help="Filename of the XML file to parse")
+    args = parser.parse_args()
+
+    print("Reading Vocabulary, this may take a minute...")
+    concept_df = vocab_maps.read_concept()  # Uncomment if you need to load a concept DataFrame
+
+    # Parse the XML file
+    tree = ET.parse(args.filename)
+
+    # Extract code elements (tag-based and attribute-based)
+    snoop_for_code_tag(tree, ".//code", ns,concept_df)
+    snoop_for_code_tag(tree, ".//*[@codeSystem]", ns,concept_df)
+
+    # Clean up the DataFrame
+    # Step 1: Remove duplicates
+    vocab_codes_cleaned = vocab_codes.drop_duplicates()
+
+    # Step 2: Sort the DataFrame by 'codeSystem'
+    vocab_codes_cleaned = vocab_codes_cleaned.sort_values(by='codeSystem')
+
+    # Pass the cleaned DataFrame into the get_code_name function to fetch concept names
+    for index, row in vocab_codes_cleaned.iterrows():
+        # If codeSystemName is missing, look it up using the codeSystem (OID)
```

Comment:
I think I went down this route and set a bad, lazy example. We don't care so much about the codeSystemName as we do a vocabulary id. Using the codeSystemName has it's problems like SNOMED CT vs SNOMED.  I should have mapped in oid.csv instead, and I think we should here as well because if avoids run-time we requests, and gives us a chance to fix things like SNOMED.

Reply:
I don't think it was bad or lazy, **but** and OMOP shall be OMOP, they change the names so 'proper' lookup is moot.
Say I have codesystem-= 2.16.840.1.113883.6.101
The CCD displayName is NUCC Health Care Provider Taxonomy
If for some reason it was null, and I looked it up via API returns
"name":"PH_ProviderCodes_NUCC",
"title":"NUCC Provider Codes"

OMOP Vocabulary for these codes is just NUCC

So I think I am just adding noise in trying to be a purist in the codesystem metadata and not OMOP centric.